### PR TITLE
New keyboard: Neito + my DZ60 keymap tweaks

### PR DESCRIPTION
**Pull request by OlliGranlund**
- Added a new upcoming keyboard, Neito
- Adjusted my previous DZ60-keymaps

### DZ60
I've fiddled around with my previous DZ60 keymaps and amde some adjustements to them. My 2 boards differ slightly and hence I've updated both to be similar to each other.

### Neito
Neito is a new Alice-like keyboard but with a ISO-enter. There has been one similar one before called "Majbritt" but this one has a slightly different layout, support for encoder and really soon a GB so you can actually get your hands on it!

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/neito/neito.c [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+#include "neito.h"
+
+bool led_update_kb(led_t led_state) {
+    bool res = led_update_user(led_state);
+    if(res) {
+        writePin(LED_CAPS_LOCK_PIN, led_state.caps_lock);
+    }
+    return res;
+}
```
Comment (drashna) [inline keyboards/neito/neito.c:25]: This isn't needed, since you have the define set for it.
```suggestion
```

File: keyboards/neito/rules.mk [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+SLEEP_LED_ENABLE = no       	# Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes            	# USB Nkey Rollover
+BACKLIGHT_ENABLE = yes       	# Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes        	# Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+ENCODER_ENABLE = yes
```
Comment (drashna) [inline keyboards/neito/rules.mk:32]:
```suggestion
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]
In general, we prefer/recommend/require that PRs be separated out, when they're different scopes (eg, keyboard vs keymap).  

I think that this should be fine here, but in the future...

**Comment by OlliGranlund** [comment]
Will keep that in mind for next time! My thought process was that it would take less effort to have the minor additional change added in this PR as it's relevant and would be easier to approve/deny. Will do those in the future in a new PR :)

**Comment by drashna** [comment]
Can definitely understand that.  But it makes it easier to review when there are fewer files.  And if something needs to be reverted...

File: keyboards/neito/rules.mk [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+COMMAND_ENABLE = no        	# Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       	# Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes            	# USB Nkey Rollover
+BACKLIGHT_ENABLE = yes       	# Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes        	# Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+ENCODER_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/neito/rules.mk:30]:
```suggestion
BOOTLOADER = atmel-dfu

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no       	# Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes
```

File: keyboards/neito/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    if (clockwise) {
+      tap_code(KC_VOLU);
+    } else {
+      tap_code(KC_VOLD);
+    }
+};
+
+bool led_update_user(led_t led_state) {
+    return true;
+}
```
Comment (fauxpark) [inline keyboards/neito/keymaps/default/keymap.c:68]:
```suggestion
```

File: keyboards/neito/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    if (clockwise) {
+      tap_code(KC_VOLU);
+    } else {
+      tap_code(KC_VOLD);
+    }
+};
+
+bool led_update_user(led_t led_state) {
+    return true;
+}
```
Comment (fauxpark) [inline keyboards/neito/keymaps/via/keymap.c:68]:
```suggestion
```

File: keyboards/neito/info.json [hunk 6]
```diff
@@ -0,0 +1,367 @@
+{
+  "keyboard_name":"Neito",
+  "url":"olli.works",
+  "maintainer":"Oliver Granlund",
+  "width":18.1875,
+  "height":14.25,
+  "matrix": {"rows": 10, "cols": 8},
```
Comment (fauxpark) [inline keyboards/neito/info.json:7]:
```suggestion
```

File: keyboards/neito/info.json [hunk 7]
```diff
@@ -0,0 +1,367 @@
...
+  "layouts":{
+     "LAYOUT":{
+        "key_count": 68,
```
Comment (fauxpark) [inline keyboards/neito/info.json:10]:
```suggestion
```

File: keyboards/keebio/iris/keymaps/olligranlund_nordic_v2/readme.md [hunk 8]
```diff
@@ -0,0 +1,8 @@
+# Iris Nordic Layout V2, now with VIA support
+### by Oliver Granlund
+
+![Finished product](https://i.imgur.com/Y1V9Dvp.jpg)
```
Comment (fauxpark) [inline keyboards/keebio/iris/keymaps/olligranlund_nordic_v2/readme.md:4]:
```suggestion
![Finished product](https://i.imgur.com/Y1V9Dvpl.jpg)
```

File: keyboards/neito/readme.md [hunk 9]
```diff
@@ -0,0 +1,16 @@
+# Neito
+### by Oliver Granlund
+
+ISO-variant of TGR Alice with a few tweaks!
+
+![First prototype](https://i.imgur.com/GbZAnIp.jpg)
```
Comment (fauxpark) [inline keyboards/neito/readme.md:6]:
```suggestion
![First prototype](https://i.imgur.com/GbZAnIpl.jpg)
```

File: keyboards/neito/readme.md [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+
+![First prototypes backside](https://i.imgur.com/h7Hx6EU.jpg)
```
Comment (fauxpark) [inline keyboards/neito/readme.md:8]:
```suggestion
![First prototypes backside](https://i.imgur.com/h7Hx6EUl.jpg)
```

File: keyboards/keebio/iris/keymaps/olligranlund_nordic_v2/config.h [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+
+// #define MASTER_LEFT
+// #define MASTER_RIGHT
+#define EE_HANDS
+
+#undef RGBLED_NUM
+
+// Via stuff?
+// #define PRODUCT_ID 3406832214
+// #define MANUFACTURER keebio
```
Comment (fauxpark) [inline keyboards/keebio/iris/keymaps/olligranlund_nordic_v2/config.h:35]:
```suggestion
```

File: keyboards/dz60/keymaps/olligranlund_iso_v2/readme.md [hunk 12]
```diff
@@ -0,0 +1,6 @@
+# DZ60 with splitted parts with VIA support
+### by Oliver Granlund
+
+![Finished product](https://i.imgur.com/AT1Lyrx.jpg)
```
Comment (fauxpark) [inline keyboards/dz60/keymaps/olligranlund_iso_v2/readme.md:4]:
```suggestion
![Finished product](https://i.imgur.com/AT1Lyrxl.jpg)
```

File: keyboards/dz60/keymaps/olligranlund_iso/readme.md [hunk 13]
```diff
@@ -1,4 +1,4 @@
-# DZ60 with splitted parts
+# DZ60 with splitted parts with VIA support
 ### by Oliver Granlund
 
 ![Finished product](https://i.imgur.com/HlEo5Yg.jpg)
```
Comment (fauxpark) [inline keyboards/dz60/keymaps/olligranlund_iso/readme.md:4]:
```suggestion
![Finished product](https://i.imgur.com/HlEo5Ygl.jpg)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/dz60/keymaps/olligranlund_iso/keymap.c [hunk 14]
```diff
@@ -19,20 +35,75 @@
...
+	EMOJI_KOVAAAJOA,
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+	switch (keycode) {
+    	case EMOJI_DANCERS:
+			if (record->event.pressed) {
+				// when keycode QMKBEST is pressed
+				SEND_STRING(">dancers>");
+			} else {} break;
```
Comment (fauxpark) [inline keyboards/dz60/keymaps/olligranlund_iso/keymap.c:51]:
```suggestion
			}
			break;
```
Don't need the else blocks if they're empty.

File: keyboards/dz60/keymaps/olligranlund_iso/keymap.c [hunk 15]
```diff
@@ -19,20 +35,75 @@
...
+
+		case EMOJI_PERJANTAI:
+			if (record->event.pressed) {
+				// when keycode QMKBEST is pressed
+				SEND_STRING(">perjantaideploy>");
+			} else {} break;
```
Comment (fauxpark) [inline keyboards/dz60/keymaps/olligranlund_iso/keymap.c:57]:
```suggestion
			}
			break;
```

File: keyboards/dz60/keymaps/olligranlund_iso/keymap.c [hunk 16]
```diff
@@ -19,20 +35,75 @@
...
+		
+		case EMOJI_THISISFINE:
+			if (record->event.pressed) {
+				// when keycode QMKBEST is pressed
+				SEND_STRING(">this/is/fine>");
+			} else {} break;
```
Comment (fauxpark) [inline keyboards/dz60/keymaps/olligranlund_iso/keymap.c:63]:
```suggestion
			}
			break;
```

File: keyboards/dz60/keymaps/olligranlund_iso/keymap.c [hunk 17]
```diff
@@ -19,20 +35,75 @@
...
+    
+		case EMOJI_KOVAAAJOA:
+			if (record->event.pressed) {
+				// when keycode QMKBEST is pressed
+				SEND_STRING(">kovaaajoa>");
+			} else {} break;
```
Comment (fauxpark) [inline keyboards/dz60/keymaps/olligranlund_iso/keymap.c:69]:
```suggestion
			}
			break;
```

File: keyboards/dz60/keymaps/olligranlund_iso_v2/keymap.c [hunk 18]
```diff
@@ -0,0 +1,109 @@
...
+	EMOJI_KOVAAAJOA,
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+	switch (keycode) {
+    	case EMOJI_DANCERS:
+			if (record->event.pressed) {
+				// when keycode QMKBEST is pressed
+				SEND_STRING(">dancers>");
+			} else {} break;
```
Comment (fauxpark) [inline keyboards/dz60/keymaps/olligranlund_iso_v2/keymap.c:51]:
```suggestion
			}
			break;
```

File: keyboards/dz60/keymaps/olligranlund_iso_v2/keymap.c [hunk 19]
```diff
@@ -0,0 +1,109 @@
...
+
+		case EMOJI_PERJANTAI:
+			if (record->event.pressed) {
+				// when keycode QMKBEST is pressed
+				SEND_STRING(">perjantaideploy>");
+			} else {} break;
```
Comment (fauxpark) [inline keyboards/dz60/keymaps/olligranlund_iso_v2/keymap.c:57]:
```suggestion
			}
			break;
```

File: keyboards/dz60/keymaps/olligranlund_iso_v2/keymap.c [hunk 20]
```diff
@@ -0,0 +1,109 @@
...
+		
+		case EMOJI_THISISFINE:
+			if (record->event.pressed) {
+				// when keycode QMKBEST is pressed
+				SEND_STRING(">this/is/fine>");
+			} else {} break;
```
Comment (fauxpark) [inline keyboards/dz60/keymaps/olligranlund_iso_v2/keymap.c:63]:
```suggestion
			}
			break;
```

File: keyboards/dz60/keymaps/olligranlund_iso_v2/keymap.c [hunk 21]
```diff
@@ -0,0 +1,109 @@
...
+    
+		case EMOJI_KOVAAAJOA:
+			if (record->event.pressed) {
+				// when keycode QMKBEST is pressed
+				SEND_STRING(">kovaaajoa>");
+			} else {} break;
```
Comment (fauxpark) [inline keyboards/dz60/keymaps/olligranlund_iso_v2/keymap.c:69]:
```suggestion
			}
			break;
```

File: keyboards/keebio/iris/keymaps/olligranlund_nordic_v2/keymap.c [hunk 22]
```diff
@@ -0,0 +1,137 @@
...
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+	switch (keycode) {
+		case QWERTY:
+			if (record->event.pressed) {
+				set_single_persistent_default_layer(_QWERTY);
+			}
+			return false;
+			break;
```
Comment (fauxpark) [inline keyboards/keebio/iris/keymaps/olligranlund_nordic_v2/keymap.c:105]:
```suggestion
```

File: keyboards/keebio/iris/keymaps/olligranlund_nordic_v2/keymap.c [hunk 23]
```diff
@@ -0,0 +1,137 @@
...
+		case LOWER:
+			if (record->event.pressed) {
+				layer_on(_LOWER);
+				update_tri_layer(_LOWER, _RAISE, _ADJUST);
+			} else {
+				layer_off(_LOWER);
+				update_tri_layer(_LOWER, _RAISE, _ADJUST);
+			}
+			return false;
+			break;
```
Comment (fauxpark) [inline keyboards/keebio/iris/keymaps/olligranlund_nordic_v2/keymap.c:115]:
```suggestion
```

File: keyboards/keebio/iris/keymaps/olligranlund_nordic_v2/keymap.c [hunk 24]
```diff
@@ -0,0 +1,137 @@
...
+		case RAISE:
+			if (record->event.pressed) {
+				layer_on(_RAISE);
+				update_tri_layer(_LOWER, _RAISE, _ADJUST);
+			} else {
+				layer_off(_RAISE);
+				update_tri_layer(_LOWER, _RAISE, _ADJUST);
+			}
+			return false;
+			break;
```
Comment (fauxpark) [inline keyboards/keebio/iris/keymaps/olligranlund_nordic_v2/keymap.c:125]:
```suggestion
```

File: keyboards/keebio/iris/keymaps/olligranlund_nordic_v2/keymap.c [hunk 25]
```diff
@@ -0,0 +1,137 @@
...
+
+		case ADJUST:
+		if (record->event.pressed) {
+			layer_on(_ADJUST);
+		} else {
+			layer_off(_ADJUST);
+		}
+		return false;
+		break;
```
Comment (fauxpark) [inline keyboards/keebio/iris/keymaps/olligranlund_nordic_v2/keymap.c:134]:
```suggestion
```

File: keyboards/neito/info.json [hunk 26]
```diff
@@ -0,0 +1,365 @@
+{
```
Comment (fauxpark) [inline keyboards/neito/info.json:1]: Please redo this file without the rotation - it currently renders like this in the Configurator (you can preview it by pressing Ctrl+Shift+I and selecting the info.json, turning on Fast Input also helps verify the ordering matches the layout macro):

image

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/neito/info.json (hunk 26 above)
Comment (fauxpark) [reply keyboards/neito/info.json:1]: This is still not quite right, there is an offset and some rounding errors.

File: keyboards/neito/rules.mk (hunk 3 above)
Comment (fauxpark) [reply keyboards/neito/rules.mk:30]: This is still outstanding.

Comment (OlliGranlund) [reply keyboards/neito/rules.mk]: Should be fixed now, sorry for the hassle :S

File: keyboards/neito/rules.mk [hunk 27]
```diff
@@ -0,0 +1,22 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+# Build Options
```
Comment (fauxpark) [inline keyboards/neito/rules.mk:6]:
```suggestion

# Build Options
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by OlliGranlund** [comment]
There were some new-line changes requested a few weeks back, I guess there is nothing pending this anymore or have I missed something?

**Review by fauxpark — APPROVED** [review]

File: keyboards/keebio/iris/keymaps/olligranlund_nordic_v2/keymap.c [hunk 28]
```diff
@@ -0,0 +1,133 @@
...
+	RAISE,
+	ADJUST,
+};
+
+#define KC_ KC_TRNS
+
+#define KC_LOWER LOWER
+#define KC_RAISE RAISE
+#define KC_RST RESET
+
```
Comment (noroadsleft) [inline keyboards/keebio/iris/keymaps/olligranlund_nordic_v2/keymap.c:38]: You don't seem to be using any of these.

```suggestion
```

File: keyboards/neito/readme.md [hunk 29]
```diff
@@ -0,0 +1,16 @@
+# Neito
+### by Oliver Granlund
+
+ISO-variant of TGR Alice with a few tweaks!
+
+![First prototype](https://i.imgur.com/GbZAnIpl.jpg)
+
+![First prototypes backside](https://i.imgur.com/h7Hx6EUl.jpg)
+
+GB:s of this board are hosted on [https://olli.works](https://olli.works)
```
Comment (noroadsleft) [inline keyboards/neito/readme.md:10]:
```suggestion
* Keyboard Maintainer: [Oliver Granlund](https://github.com/OlliGranlund)
* Hardware Supported: Neito PCB (ATmega32U4)
* Hardware Availability: Group Buys hosted on [https://olli.works](https://olli.works)
```

File: keyboards/neito/readme.md [hunk 30]
```diff
@@ -0,0 +1,16 @@
...
+
+Make example for this keyboard (after setting up your build environment):
+
+    make neito:default
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```
Comment (noroadsleft) [inline keyboards/neito/readme.md:16]: Instructions on how to reset the board to bootloader mode should go above this line.

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Comment by OlliGranlund** [comment]
The implementation of Neito-keyboard in this PR works fine, but I've cleaned this up and made its matrix more clear. If I were to push the change here, it wouldn't be fair to you so I guess I'm waiting for this to be merged so that I can make another PR for the improvements.

https://github.com/OlliGranlund/qmk_firmware/tree/neito-fixed

I also thought about making an own PR for **neito-fixed** but I fear that it'll just be double work for you and has a risk of messing things up. I guess it's best to wait that this is merged?

**Comment by drashna** [comment]
Honestly, I'd prefer the changes be added here.  From the looks of it, it's not too much, so should be fine.

**Comment by OlliGranlund** [comment]
I've merge my branch into this PR. The Neito default and via keymaps are really baseline and I separate my own "optimized" layout to it's own keymap folder.

File: keyboards/neito/rules.mk [hunk 31]
```diff
@@ -0,0 +1,24 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+ENCODER_ENABLE = yes		# We have a encoder
+MOUSEKEY_ENABLE = yes		# In case someone wants to use mousekeys and doesn't want to reflash stock firmware
```
Comment (fauxpark) [inline keyboards/neito/rules.mk:24]:
```suggestion
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes        # We have a encoder
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/neito/keymaps/olli.works/keymap.c [hunk 32]
```diff
@@ -0,0 +1,64 @@
+/* Copyright 2020 Oliver Granlund
```
Comment (fauxpark) [inline keyboards/neito/keymaps/olli.works/keymap.c:1]: The `.` in this keymap name may need to be changed to an `_`, I'm not sure how well the build system/CLI will take it.

**Comment by OlliGranlund** [comment]
I think git has become confused as the proposed changes have been made but it still says there are some requested things left to handle.

**Comment by OlliGranlund** [comment]
This PR has been up for nearly 4 months. I wonder what else there is to be done to get it merged? The requested changes by Github seems bugged as I've done the changes already.

**Review by drashna — APPROVED** [review]

**Comment by OlliGranlund** [comment]
Updated the Neito encoders to support the latest version of QMK, hopefully this will help with approving this PR :)

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**